### PR TITLE
07 post comment

### PR DESCRIPTION
**Pull request by Amir-Rsh**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -65,3 +65,90 @@ describe("GET /api/articles/:article_id", () => {
...
+  });
+  it("400: responds with appropriate message when posted comment contains invalid property values", async () => {
+    const newComment = { username: 8777, body: 67 };
+    const response = await request(app)
+      .post("/api/articles/10/comments")
+      .send(newComment);
+    expect(response.status).toBe(400);
+    expect(response.body.msg).toBe("Bad Request");
+  });
+});
```
Comment (nc-richard-neat) [inline __tests__/app.test.js:154]: Nice testing here, we could also consider:
- 400 for invalid id
- 404 for username doesn't exist

File: controllers/comments.controller.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  } catch (err) {
+    next(err);
+  }
+};
+
+exports.postComments = async (req, res, next) => {
+  try {
+    const newComment = req.body;
+    const { article_id } = req.params;
+    const checkArticle = await checkArticleExists(article_id);
```
Comment (nc-richard-neat) [inline controllers/comments.controller.js:21]: We likely won't need to check the article exists first, as this is a foreign key psql will throw an error if you try to insert onto an article that doesn't exist.

**Review by nc-richard-neat — COMMENTED** [review]
Nice work on this, some suggestions in the comments.
Some advice regarding general commit conventions: there shouldn't be any code related to previous tasks on this commit, keep all of the changes only relevant to the current task 👍

File: __tests__/app.test.js (hunk 1 above)
Comment (Amir-Rsh) [reply __tests__/app.test.js:154]: Thanks for the feedback Richard. Apologies for the mix-up with the previous tasks. Would it be ok if i apply the feedback on 'main' and then create the new tickets?

Comment (nc-richard-neat) [reply __tests__/app.test.js:154]: Once you've received the feedback that's up to you how to implement it. It's always good to use branching tools to keep our commits and code organised though :)

**Pull request merged**